opendg/_storage/base.py

```diff
@@ -48,14 +38,18 @@ def temporal_coarsening(
     ) -> 'DGStorageBase':
         r"""Re-index the temporal axis of the dynamic graph."""
 
+    def __len__(self) -> int:
+        r"""Returns the number of temporal length of the dynamic graph."""
+        return self.num_timestamps
+
     @property
     @abstractmethod
-    def start_time(self) -> int:
+    def start_time(self) -> Optional[int]:
         r"""The start time of the temporal graph."""
 
     @property
     @abstractmethod
-    def end_time(self) -> int:
+    def end_time(self) -> Optional[int]:
```

Comment:
We need to decide upon certain graph semantics. I'm assuming we allow empty graphs. In that case, start/end time are undefined (`None`).